pkg/execution/executor/executor.go

```diff
@@ -834,16 +834,15 @@ func (e *executor) Cancel(ctx context.Context, id state.Identifier, r execution.
 		return nil
 	}
 
-	// TODO: Load all pauses for the function and remove.
-
 	if err := e.sm.Cancel(ctx, md.Identifier); err != nil {
 		return fmt.Errorf("error cancelling function: %w", err)
 	}
 
-	s, _ := e.sm.Load(ctx, id.RunID)
+	// TODO: Load all pauses for the function and remove, once we index pauses.
 
+	s, _ := e.sm.Load(ctx, runID)
```

Comment:
no error handling here?

Reply:
I only regorganized comments with this code.  AFAIK, state is a nice to have with this lifecycle.